homeassistant/components/enocean/sensor.py

```diff
@@ -159,6 +160,11 @@ class EnOceanPowerSensor(EnOceanSensor):
     - A5-12-01 (Automated Meter Reading, Electricity)
     """
 
+    def __init__(self, dev_id, dev_name, description: SensorEntityDescription):
+        """Initialize the EnOcean power sensor device."""
+        super().__init__(dev_id, dev_name, description)
+        self._attr_unique_id = f"{combine_hex(dev_id)}-{SENSOR_TYPE_POWER}"
```

Comment:
Add an attribute in an entity description mixin class that accepts the device id and returns the unique_id.

Example:
https://github.com/home-assistant/core/blob/a70e2a33dcd85608f1145d8fc2e89a87620f4ef3/homeassistant/components/wled/sensor.py#L34-L47